src/server/common/dbconn.cpp

```diff
@@ -292,58 +316,57 @@ int dbconn::get_single_int(const std::string& sql, Args&&... args)
 			case mariadb::value::type::unsigned32:
 			case mariadb::value::type::signed32:
 				return rslt->get_signed32(0);
+			case mariadb::value::type::unsigned64:
+			case mariadb::value::type::signed64:
+				return rslt->get_signed64(0);
 			default:
-				throw mariadb::exception::base("Value retrieved was not an int!");
+				throw mariadb::exception::base("Value retrieved was not a long!");
 		}
 	}
 	else
 	{
-		throw mariadb::exception::base("No int value found in the database!");
+		throw mariadb::exception::base("No long value found in the database!");
 	}
 }
 template<typename... Args>
-bool dbconn::exists(const std::string& sql, Args&&... args)
+bool dbconn::exists(const std::string& sql, bool async, Args&&... args)
 {
-	mariadb::result_set_ref rslt = select(sql, args...);
+	mariadb::result_set_ref rslt = select(sql, async, args...);
 	return rslt->next();
 }
 
 //
 // select or modify values
 //
 template<typename... Args>
-mariadb::result_set_ref dbconn::select(const std::string& sql, Args&&... args)
+mariadb::result_set_ref dbconn::select(const std::string& sql, bool async, Args&&... args)
 {
 	try
 	{
-		mariadb::statement_ref stmt = query(sql, args...);
-		return mariadb::result_set_ref(stmt->query());
+		mariadb::connection_ref conn = async ? get_conn() : connection_;
+		mariadb::statement_ref stmt = query(sql, conn, args...);
+		mariadb::result_set_ref rslt = mariadb::result_set_ref(stmt->query());
+		return rslt;
 	}
 	catch(const mariadb::exception::base& e)
 	{
-		if(!connection_->connected())
-		{
-			ERR_SQL << "Connection is invalid!" << std::endl;
-		}
-		ERR_SQL << "SQL query failed for query: `"+sql+"`!" << std::endl;
+		ERR_SQL << "SQL query failed for query: `"+sql+"` on thread: " << std::this_thread::get_id() << std::endl;
```

Comment:
Since the thread-safety support is being introduced into the central logging i'd rather add an option there to include the thread ID in every log message.

Reply:
Every log message in dbconn, or do you mean something more wide-ranging?

Reply:
Every log message at all. I want to know what happens in which thread on the client as well. The format should be concise though so just an additional number (padded to 5 digits or so) somewhere at the start of the log message.

Reply:
Alright.